### PR TITLE
feat: Add permissions filter to EditGroup and EditSpacePermissions

### PR DESCRIPTION
**Pull request by Rockyy174**
I've added a permissions filter to the `OperatelyWeb.Api.Mutations.EditGroup` and `OperatelyWeb.Api.Mutations.EditSpacePermissions` mutations.

### CODE REVIEW

**Comment by Rockyy174** [comment]
I've implemented two variants of the mutation handler: 
1) One using `case` and `if` statements
2) Another using a railway approach

To be honest, I'm not sure which one I prefer. What do you think, @shiroyasha?

I will also implement a 3) one using `with` statement, but I will have to modify the `Filters` and `Permissions` module, so I will submit it separately as draft.

File: lib/operately_web/api/mutations/edit_group.ex [hunk 1]
```diff
@@ -13,14 +16,22 @@ defmodule OperatelyWeb.Api.Mutations.EditGroup do
...
 
   def call(conn, inputs) do
+    person = me(conn)
     {:ok, space_id} = decode_id(inputs.id)
-    space = Operately.Groups.get_group!(space_id)
 
-    {:ok, space} = Operately.Groups.edit_group_name_and_purpose(me(conn), space, %{
-      name: inputs.name,
-      mission: inputs.mission
-    })
```
Comment (Rockyy174) [inline lib/operately_web/api/mutations/edit_group.ex:22]: Using `case` and `if`

File: lib/operately_web/api/mutations/edit_group.ex [hunk 2]
```diff
@@ -13,14 +16,22 @@ defmodule OperatelyWeb.Api.Mutations.EditGroup do
...
+    case Groups.get_group_and_access_level(space_id, person.id) do
+      {:ok, space, access_level} ->
+        if Permissions.can_edit(access_level) do
+          execute(person, space, inputs)
```
Comment (Rockyy174) [inline lib/operately_web/api/mutations/edit_group.ex:25]: Using `case` and `if`

File: lib/operately_web/api/mutations/edit_space_permissions.ex [hunk 3]
```diff
@@ -12,12 +15,34 @@ defmodule OperatelyWeb.Api.Mutations.EditSpacePermissions do
   end
 
   def call(conn, inputs) do
-    {:ok, id} = decode_id(inputs.space_id)
+    person = me(conn)
+
+    load_group(inputs, person)
+    |> verify_permissions()
+    |> execute(person, inputs)
+  end
```
Comment (Rockyy174) [inline lib/operately_web/api/mutations/edit_space_permissions.ex:23]: Second approach

File: lib/operately_web/api/mutations/edit_group.ex (hunk 2 above)
Comment (shiroyasha) [reply lib/operately_web/api/mutations/edit_group.ex:25]: This one looks quite clean, but if a function is returning {:ok, space, access_level} i would expect the alternative answer to be {:error, reason}.

For example:

```
case Groups.get_group_and_access_level(space_id, person.id) do
  {:ok, space, access_level} ->
  {:error, :not_found} -> ...
end
```

File: lib/operately_web/api/mutations/edit_space_permissions.ex (hunk 3 above)
Comment (shiroyasha) [reply lib/operately_web/api/mutations/edit_space_permissions.ex:23]: This one is nice, but the fact that we need to implement multiple alternative function is PITA.

File: lib/operately_web/api/mutations/edit_group.ex (hunk 2 above)
Comment (Rockyy174) [reply lib/operately_web/api/mutations/edit_group.ex:25]: Makes sense

**Comment by Rockyy174** [comment]
I've updated both EditGroup and EditSpacePermissions to use the solution with the `case` and `if` statements. Let me know if you have any other improvements to suggest!

**Comment by shiroyasha** [comment]
Looks great! Nothing else for changing.

**Pull request merged**